### DISCUSSION TITLE
Debug highlighted command using IDE not running in debug

### DISCUSSION THREAD

**Discussion by andrewring** [intellij-rust/intellij-rust#8778] [IntelliJ Rust Plugin Development]
The ability to quickly run a terminal command in the IDE, using Ctrl+Enter, is very handy. It would be even better if I could kick off in IDE debug that way... and sure enough it's a feature included in CLion, using Ctrl+Shift+Enter. Unfortunately, running a cargo command this way does not, in fact, run it as a Debug profile, but simply as a Run. The run command can be copied to Debug manually, but it would be great to have this working correctly.

**Comment by Undin** [comment]
@andrewring Are you about [Run Anything](https://www.jetbrains.com/help/clion/running-anything.html) feature?
If so, debugging support was implemented in #8697. You can try it right now using [nightly](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-quick-start.html#install-nightly) builds or wait for stable builds next week

**Reply by andrewring** [reply]
No, not the Run Everything, this is from the terminal. Typing out a cargo command in the in IDE terminal auto highlights, and then Ctrl+Enter switches to the Run tab, and runs the command there.

**Reply by Undin** [reply]
Got it! Actually, this feature is based on `Run Anything` implementation under the hood (at least, it reuses the corresponding implementation) so #8697 makes this feature work as expected as well

**Reply by andrewring** [reply]
Oh, fantastic, thank you!
